### PR TITLE
build: ensure msrv 1.80.1 and upgrade dependencies

### PR DESCRIPTION
**Pull request by tisonkun**
This refers to #8

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -107,14 +108,18 @@ fn main() {
...
-| fastrace (baseline) | 1x (0.4us)  | 1x (0.8us) | 1x (3.4us)  | 1x (27.8us) |
+|                     | 1 span     | 10 spans   | 100 spans   | 1000 spans  |
+|---------------------|------------|------------|-------------|-------------|
+| tokio-tracing       | 19x slower | 61x slower | 124x slower | 151x slower |
+| rustracing          | 13x slower | 26x slower | 45x slower  | 55x slower  |
+| fastrace (baseline) | 1x (0.4us) | 1x (0.8us) | 1x (3.4us)  | 1x (27.8us) |
 
 Detailed results are available in [etc/benchmark-result](etc/benchmark-result).
 
+## Supported Rust Versions (MSRV 1.71)
```
Comment (Xuanwo) [inline README.md:119]: Should be `MSRV 1.80.1`?

File: README.md [hunk 2]
```diff
@@ -107,14 +108,18 @@ fn main() {
...
+
+Fastrace is built against the latest stable release. The minimum supported version is 1.80.1. The current Fastrace version is not guaranteed to build on Rust versions earlier than the minimum supported version.
```
Comment (Xuanwo) [inline README.md:121]: I feel like it's too aggressive to have an MSRV like this. But we can see if it works well and evaluate it as needed.

**Review by Xuanwo — COMMENTED** [review]
Thanks!

File: README.md (hunk 1 above)
Comment (tisonkun) [reply README.md:119]: Yes.

File: README.md (hunk 2 above)
Comment (tisonkun) [reply README.md:121]: The background is that, we don't actually define an MSRV before and it's effectively latest stable. Thus, use the current stable version is the least surprising one.

File: README.md (hunk 1 above)
Comment (tisonkun) [reply README.md:119]:
```suggestion
## Supported Rust Versions (MSRV 1.80.1)
```

File: README.md (hunk 2 above)
Comment (tisonkun) [reply README.md:121]: Also reducing MSRV should not be a breaking change IIUC.

Comment (Xuanwo) [reply README.md:121]: I'm fine with this policy.

**Review by andylokandy — APPROVED** [review]

**Comment by andylokandy** [comment]
```
thread 'test_macro_properties' panicked at fastrace/tests/lib.rs:758:5:
assertion `left == right` failed
  left: "\nroot []\n    bar_async []\n    foo_async [(\"k1\", \"v1\"), (\"a\", \"argument a is 1\"), (\"b\", \"Bar\"), (\"escaped1\", \"Bar{}\"), (\"escaped2\", \"{ \\\"a\\\": \\\"b\\\"}\")]\n"
 right: "\nroot []\n    bar []\n    bar_async []\n    foo [(\"k1\", \"v1\"), (\"a\", \"argument a is 1\"), (\"b\", \"Bar\"), (\"escaped1\", \"Bar{}\"), (\"escaped2\", \"{ \\\"a\\\": \\\"b\\\"}\")]\n    foo_async [(\"k1\", \"v1\"), (\"a\", \"argument a is 1\"), (\"b\", \"Bar\"), (\"escaped1\", \"Bar{}\"), (\"escaped2\", \"{ \\\"a\\\": \\\"b\\\"}\")]\n"
 ```
 Seems the same problem as https://github.com/fastracelabs/fastrace/issues/9 but on linux.

**Comment by tisonkun** [comment]
@andylokandy Then perhaps merge this one and debugging the macro issue later?

**Comment by andylokandy** [comment]
> Then perhaps merge this one and debugging the macro issue later?

Yes, was just waiting for ci to rerun

**Pull request merged**